src/rest/handlers/rest_vertices.erl

```diff
@@ -105,7 +107,62 @@ to_json(Req, State = #{operation := listing}) ->
     {ok, Vertices} = graph:list_vertices(),
     {gmm_utils:encode(Vertices), Req, State}.
 
-
+%%% BULK handler
+-spec replace(T, T, [T]) -> [T].
+replace(Element, Replacement, [Element | T]) ->
+    [Replacement | T];
+replace(Element, Replacement, [H | T]) ->
+    [H | replace(Element, Replacement, T)];
+replace(_Element, _Replacement, []) ->
+    [].
+-spec modify_state_bulk(Vertices :: list({binary(), binary()})) -> ok | {error, any()}.
+modify_state_bulk(Vertices)->
+    Parent = self(),
+    Ref = erlang:make_ref(),
+
+    Pids = lists:map(fun({Type, Name}) ->
+        spawn(fun() ->
+            Result = try
+                         graph:create_vertex(Type, Name)
+                     catch Type:Reason:Stacktrace ->
+                {'$pmap_error', self(), Type, Reason, Stacktrace}
+                     end,
+            Parent ! {Ref, self(), Result}
+              end)
+                     end, Vertices),
+
+    % GATHERING RESULTS
+    Gather = fun
+    %PidsOrResults is initially the list of pids, gradually replaced by corresponding results
+                 F(PendingPids = [_ | _], PidsOrResults) ->
+                     receive
+                         {Ref, Pid, Result} ->
+                             NewPidsOrResults = replace(Pid, Result, PidsOrResults),
+                             F(lists:delete(Pid, PendingPids), NewPidsOrResults)
+                     after 5000 ->
+                         case lists:any(fun erlang:is_process_alive/1, PendingPids) of
```

Comment:
Should it not be: ..?
```suggestion
                         case lists:all(fun erlang:is_process_alive/1, PendingPids) of
```

It's up to debate, but I would advice to reconsider it.